mne/inverse_sparse/_gamma_map.py

```diff
@@ -290,21 +292,25 @@ def gamma_map(evoked, forward, noise_cov, alpha, loose="auto", depth=0.8,
         in_pos = 0
         if len(X) < 3 * len(active_src):
             X_xyz = np.zeros((3 * len(active_src), X.shape[1]), dtype=X.dtype)
+            X_ori_ = np.zeros((3 * len(active_src), X.shape[1]), dtype=X.dtype)
             for ii in range(len(active_src)):
                 for jj in range(3):
                     if in_pos >= len(active_set):
                         break
                     if (active_set[in_pos] + jj) % 3 == 0:
                         X_xyz[3 * ii + jj] = X[in_pos]
+                        X_ori_[3 * ii + jj] = X_ori[in_pos]
                         in_pos += 1
             X = X_xyz
+            X_ori = X_ori_
 
     tmin = evoked.times[0]
     tstep = 1.0 / evoked.info['sfreq']
 
     if return_as_dipoles:
         out = _make_dipoles_sparse(X, active_set, forward, tmin, tstep, M,
-                                   M_estimated, active_is_idx=True)
+                                   M_estimated, active_is_idx=True,
+                                   X_ori=X_ori)
```

Comment:
it's really a mystery to me why you need to pass both X and X_ori and why internally you use X to get amplitude and X_ori to get orientation.